core/domain/user_jobs_one_off.py

```diff
@@ -468,6 +469,325 @@ def reduce(key, values):
+        exploration_user_data_model = (
+            user_models.ExplorationUserDataModel.query(
+                user_models.ExplorationUserDataModel.user_id == user_id).get()
+        )
+
+        all_models_linked_with_user_settings_model = [
+            ('ExplorationUserDataModel', exploration_user_data_model)
+        ]
+
+        # Models in user storage module keyed by user_id itself.
+        model_names_and_ids_to_be_fetched_in_batch = [
+            ('UserContributionsModel', [user_id]),
+            ('UserEmailPreferencesModel', [user_id]),
+            ('UserStatsModel', [user_id]),
+            ('UserSubscriptionsModel', [user_id]),
+        ]
+        fetched_batch_models = (
+            datastore_services.fetch_multiple_entities_by_ids_and_models(
+                model_names_and_ids_to_be_fetched_in_batch)
+        )
+
+        for model_name_tuple, model_list in list(python_utils.ZIP(
+                model_names_and_ids_to_be_fetched_in_batch,
+                fetched_batch_models)):
+            model_name = model_name_tuple[0]
+            actual_model = model_list[0]
+            all_models_linked_with_user_settings_model.append(
+                (model_name, actual_model)
+            )
+
+        for model_name, model in all_models_linked_with_user_settings_model:
+            if model is not None:
+                user_dates_list.append(
+                    (
+                        model_name + python_utils.UNICODE('_last_updated'),
+                        model.last_updated
+                    )
+                )
+                user_dates_list.append(
+                    (
+                        model_name + python_utils.UNICODE('_created_on'),
+                        model.created_on
+                    )
+                )
+                if model_name == 'UserSubscriptionsModel':
+                    user_dates_list.append(
+                        (
+                            'UserSubscriptionsModel_last_checked',
+                            model.last_checked
+                        )
+                    )
+                if model_name == 'ExplorationUserDataModel':
+                    user_dates_list.append(
+                        (
+                            'ExplorationUserDataModel_rated_on',
+                            model.rated_on
+                        )
+                    )
+                    user_dates_list.append(
+                        (
+                            'ExplorationUserDataModel_draft_change_list_last_'
+                            'updated',
+                            model.draft_change_list_last_updated
+                        )
+                    )
+
+        filtered_user_dates_list = [
+            (attribute_name, date) for attribute_name, date in user_dates_list
+            if date is not None
+        ]
+        model_name, min_date = min(filtered_user_dates_list, key=lambda x: x[1])
```

Comment:
@seanlip I think this is important so I will describe it in a bit detail. 

Here my assumption was that `UserSettingsModel.last_updated` would be present for all models. (I believe `last_agreed_to_terms` and `last_logged_in` should also be at least present, though I was unsure if there exists a scenario where it's possible that one or both of these might not exist). 

With this assumption only I proceeded to exclude `UserSettingsModel.created_on` itself from the list. Though I realise now I could have still included it in the list, without adding unwanted updates and added complexity overall, and make the overall solution more safe if my assumption isn't always absolutely true.

If you see any problem with these assumptions, I can go ahead and do further analysis, or include `UserSettinhgsModel.created_on` in the list which makes the solution foolproof.

Reply:
Thanks for explaining! I agree with your analysis. I am not too worried though because I didn't see any issues in my test run and all indications suggest that everything was fine for users created within the last 6 months, so I think this is just a theoretical concern.

If you want to be safe, it's fine to do a follow-up PR, but I think it's probably optional. You might want to get @vojtechjelinek's opinion.